core/src/main/scala/kafka/coordinator/ConsumerMetadata.scala

```diff
@@ -46,4 +46,5 @@ private[coordinator] class ConsumerMetadata(val consumerId: String,
   var awaitingRebalanceCallback: (Set[TopicAndPartition], String, Int, Short) => Unit = null
   var assignedTopicPartitions = Set.empty[TopicAndPartition]
   var latestHeartbeat: Long = -1
+  var isLeaving: Boolean = false
```

Comment:
Why do we need this flag if we always remove the consumer from HB purgatory immediately upon leave group? Since heartbeatPurgatory.checkAndComplete() is synchronized with expire() it should not be accessed concurrently right?